kong/api/routes/upstreams.lua

```diff
@@ -166,6 +180,16 @@ return {
                                             "upstream",
                                             "page_for_upstream"),
     POST = function(self, db)
+      -- updating a target using POST is a compatibility with existent API and
+      -- should be deprecated in next major version
+      local entity, _, err_t = update_existent_target(self, db)
+      if err_t then
+        return endpoints.handle_error(err_t)
+      end
+      if entity then
+        return kong.response.exit(200, entity)
```

Comment:
Maybe we should add a `Deprecation` header like https://github.com/Kong/kong/pull/6465/files#diff-4d5b2a7046fe9022c548c4212bfc5ae9596c21237771d0e628a319fab2597785R53.

Reply:
Nice catch, thanks! I jusrt changed it.